src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiExtension.groovy

```diff
@@ -92,6 +95,9 @@ class JpiExtension implements JpiExtensionBridge {
         this.generateTests = project.objects.property(Boolean).convention(false)
         this.requireEscapeByDefaultInJelly = project.objects.property(Boolean).convention(true)
         this.generatedTestClassName = project.objects.property(String).convention('InjectedTest')
+        this.checkstyleEnabled = project.objects.property(Boolean).convention(false)
+        this.jacocoEnabled = project.objects.property(Boolean).convention(false)
+        this.spotBugsEnabled = project.objects.property(Boolean).convention(false)
```

Comment:
Do you think it makes sense to have a higher level property to allow opting into all the OSS recommendations? It's really nice to have this level of granularity, but I'm wondering if folks will basically want two modes most of the time - one for internal plugins, and one for oss plugins.

Reply:
That's a really good point. Maybe it eventually makes sense to split into two plugins (oss, internal) for the different use-cases, but this is a good starting point and we can see where it evolves to. Happy to move forward with this.